File: src/sdk/php/3/core-classes/memory-storage/geoadd/snippets/geoadd-1.php

<?php

use \Kuzzle\Kuzzle;

$kuzzle = new Kuzzle('localhost');

$points = [
  [
    'lon' => 13.361389,
    'lat' => 38.115556,
    'name' => 'Palermo'
  ],
  [
    'lon' => 15.087269,
    'lat' => 37.502669,
    'name' => 'Catania'
  ]
];

try {
  $count = $kuzzle->memoryStorage()->geoadd('key', points);
}
catch (ErrorException $e) {

}

File: src/sdk/java/2/core-classes/memory-storage/geoadd/snippets/geoadd-1.java

JSONObject[] points = new JSONObject[]{
  new JSONObject()
    .put("lon", 13.361389)
    .put("lat", 38.115556)
    .put("name", "Palermo"),
  new JSONObject()
    .put("lon", 15.087269)
    .put("lat", 37.502669)
    .put("name", "Catania")
};

kuzzle.memoryStorage.geoadd("key", points, new ResponseListener<Long>() {
  @Override
  public void onSuccess(int count) {
    // callback called once the action has completed
  }

  @Override
  public void onError(JSONObject error) {
  }
});
